client/dashboard/data/domain-dns-records.ts

```diff
@@ -0,0 +1,42 @@
+import wpcom from 'calypso/lib/wp';
+
+export type DNSRecordType = 'A' | 'AAAA' | 'ALIAS' | 'CAA' | 'CNAME' | 'MX' | 'NS' | 'SRV' | 'TXT';
+
+export type DNSRecord = {
+	type?: DNSRecordType;
+	name: string;
+	flags?: number;
+	tag?: string;
+	value?: string;
+	target?: string;
+	data?: string;
+	weight?: number;
+	port?: number;
+	aux?: number;
+	service?: string;
+	protocol?: string;
+	ttl: number;
+};
+
+export type AddDNSRecordResponse = {
+	records: DNSRecord[];
+};
+
+export async function addDNSRecord(
```

Comment:
Do you think we should have a specific function or just have a more generic update mutation? (I have it in my PR)

Reply:
Yeah, I think that makes sense. I'll update the code with the one in Igor's PR.